src/tunnel.cc

```diff
@@ -951,6 +966,9 @@ TunnelStateData::tunnelEstablishmentDone(Http::TunnelerAnswer &answer)
 
     al->cache.code.update(LOG_TCP_TUNNEL);
 
+    // XXX: al->hier.peer_reply_status (i.e. *status_ptr) should not be (re)set
+    // until we actually start responding to the client. Right here/now, we only
+    // know how this cache_peer has responded to us.
     if (answer.peerResponseStatus != Http::scNone)
         *status_ptr = answer.peerResponseStatus;
```

Comment:
Addressing this TODO requires quite a few of out-of-scope changes with difficult-to-predict outcomes, including access logging changes. It deserves a dedicated PR.